central/sensor/augmentation/broker.go

```diff
@@ -0,0 +1,88 @@
+package augmentation
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/sensor/service/connection"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// The Broker coordinates and matches deployment enhancement requests to responses
+type Broker struct {
+	requests map[string]chan<- *central.DeploymentEnhancementResponse
+	lock     sync.Mutex
+	once     sync.Once
+}
+
+// NewBroker returns a new broker
+func NewBroker() *Broker {
+	return &Broker{
+		requests: make(map[string]chan<- *central.DeploymentEnhancementResponse),
+	}
+}
+
+// NotifyDeploymentReceived matches the ID of Sensors response to the request and notifies the waiting goroutine
+func (b *Broker) NotifyDeploymentReceived(msg *central.DeploymentEnhancementResponse) {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	if msg.GetMsg() == nil {
+		log.Warnf("Received empty message, skipping augmentation notify")
+		return
+	}
+	if r, ok := b.requests[msg.GetMsg().GetId()]; ok {
+		// Once, to prevent writing to a closed channel if a msg dupe arrives
+		b.once.Do(func() {
+			log.Debugf("Received answer for Deployment enrichment requestID %v", msg.GetMsg().GetId())
+			r <- msg
+			close(r)
+		})
```

Comment:
This is unfortunately a bad smell. If there are some unexpected messages possible, we should not close the channel immediately after sending.

Reply:
Maybe we shouldn't close these channels at all and only remove them from the map. GC will take care of them if they are unreachable. 

There shouldn't be two messages written to this channel ever. Since this function is within a lock, we should simply delete the key from the map if was found. The second (incorrect) message should have `ok = false` when checking the map, and we could log a warning/error if that ever happens.